src/artifacts.js

```diff
@@ -254,23 +325,85 @@ api.declare({
         });
     }
 
+    if (artifact.storageType === 'blob') {
```

Comment:
We need to modify the `_.isEqual` above... so that it ignores the `uploadId`...

If we have two concurrent calls to `createArtifact` and both of them creates initiates upload and creates the artifact entity we just drop the last one of the `uploadId`s...

Note: I find this rather extremely unlikely happen... It could happen if the `createArtifact` call takes more than 30s and the worker, hence, retries the request. So mostly I care that we handle the case in a consistent manner. Cancelling the `uploadId` would be nice, but how often is going to happen that we run into this... I'm guessing we can ignore cancellation and just let the uploadId time out in that instance...